projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js

```diff
@@ -0,0 +1,175 @@
+				data: createData( theData, colors[ i ], name ),
+			} );
+			currentCount = currentCount + theData;
+		}
+
+		// sort & split items into included and other
+		const PostTypeItems = splitUsablePostTypes( postTypeBreakdown, numItems, maxRecordCount );
+
+		// push includedItems into the recordInfo
+		for ( const item in PostTypeItems.includedItems ) {
+			recordInfo.push( {
+				data: createData(
+					PostTypeItems.includedItems[ item ].data.data[ 0 ],
+					colors[ item ],
+					PostTypeItems.includedItems[ item ].data.label
+				),
+			} );
+		}
+
+		// populate the 'other' category with combined remaining items and push to end of data array
+		if ( PostTypeItems.otherItems.length > 0 ) {
+			recordInfo.push( {
+				data: createData(
+					combineOtherCount( PostTypeItems.otherItems ),
+					PALETTE.colors[ 'Gray 30' ],
+					'Other'
+				),
+			} );
+		}
+
+		// if there is remaining unused space in tier, add filler spacing to chart
+		if ( tier - currentCount > 0 ) {
+			recordInfo.push( {
+				data: createData( tier - currentCount, PALETTE.colors[ 'Gray 0' ], 'Remaining' ),
+			} );
+		}
+	}
+
+	// set a var to check all the data is valid (this helps determine whether to output a chart)
+	const isValid = hasBeenIndexed && hasValidData && hasItems;
+
+	return {
+		data: recordInfo,
+		tier: tier,
+		recordCount: currentCount,
+		hasBeenIndexed,
+		hasValidData,
+		hasItems,
+		isValid,
+	};
+}
+
+/**
+ * split post types into those being displayed,
+ * and those combined into the 'other' category
+ *
+ * @param {Array} postTypeBreakdown - an array of the different post types with their counts
+ * @param {number} numItems - count of different post types
+ * @param {number} maxRecordCount - the max number of records to display before combining the rest into the 'other' category
+ * @returns {object} containing included items with post type and count, and other items, split.
+ */
+export function splitUsablePostTypes( postTypeBreakdown, numItems, maxRecordCount ) {
+	postTypeBreakdown.sort( ( a, b ) => ( a.data.data[ 0 ] < b.data.data[ 0 ] ? 1 : -1 ) );
+
+	const count = maxRecordCount <= numItems ? maxRecordCount : numItems;
+
+	return {
+		includedItems: postTypeBreakdown.slice( 0, count ),
+		otherItems: postTypeBreakdown.slice( count, numItems ),
+	};
+}
+
+/**
+ * combine remaining item count for use in 'other' category
+ *
+ * @param {Array} otherItems - array of items, with their counts, to be combined into 'other'
+ * @returns {number} sum of all 'other' item counts
+ */
+export function combineOtherCount( otherItems ) {
+	let runningTotal = 0;
+
+	for ( const item in otherItems ) {
+		runningTotal = otherItems[ item ].data.data[ 0 ] + runningTotal;
+	}
+
+	return runningTotal;
+}
+
+/**
+ * capitalize the first letter of any string
+ *
+ * @param {string} string - takes any string
+ * @returns {string} the same string, with the first letter capitalized
+ */
+export function capitalizeFirstLetter( string ) {
```

Comment:
Just a note that we might have to remove this later - it likely won't work across all languages.

Reply:
Yep I agree. Will have a think about a good alternative.